### ISSUE TITLE
"no topic set" regression

### ISSUE THREAD

**Issue by dequis** [irssi/irssi#903]
Received this on staff@irssi.org, haven't looked into it:

> Hello.
> 
> Upon joining the channel, it says:
> 
> 11:16:25 -!- Topic for ##austrian-economics: Welcome to the unofficial chat of the Ludwig von Mises Institute! Be warned: no censorship applies to this channel, and off-topic conversations may be present.
> 11:16:25 -!- Topic set by zolk3ri [~zolk3ri@gateway/tor-sasl/zolk3ri] [Thu Aug  9 11:01:33 2018]
> 
> but when I type /topic, it shows:
> 
> 11:55:08 -!- No topic set for ##austrian-economics
> 
> And it also doesn't show up on the top bar.
> 
> I am using Irssi v1.2-g0c94d44f.
> 
> Kind Regards,
> Zoltán Kéri

Related #897 #888

**Comment by horgh** [comment]
Yeah, it's because of #897. `channel_change_topic()` gets called for both numerics 332 (topic) and 333 (topic info). Sorry about that.

In the call from 332 we have the topic text. In the call from 333 we have setby and settime. 333 typically happens after, so the topic gets lost.

One solution may be to call a different function for these two numerics so we could handle them separately. That might be clean.

**Comment by horgh** [comment]
horgh/irssi@a5444ec21169fdcbaeb34843cbc52a03f2fe8d37 is an example of what I mean. I need to test and think about it some more as I just whipped it up quickly so I won't make a PR yet, but if anyone has feedback that would be cool!

**Comment by dequis** [comment]
Hmm. I guess showing an empty topic was a feature in a way, since it also allowed you to know who unset it and when. That was also lost as part of this change. Maybe we could revert the other change and just make it display the empty topic line as unset instead?

**Comment by bmeh** [comment]
I've tried the same version (v1.2-g0c94d44f) and /topic <TAB> doesn't work either. IIRC it was somewhat random in earlier versions as well, sometimes it worked, and sometimes it didn't work. I don't know why.

I would like to be able to see the topic on the top bar at the very least though, and be able to print it through the command /topic. /topic <TAB> was useful (as an operator) for quickly changing a part of the topic without altering the rest of it. Any ideas on that?

**Comment by dequis** [comment]
Revert to an earlier version?

**Comment by horgh** [comment]
Yeah, there's an argument that knowing when it was unset could be nice. If we did want that, we might want to change the text that says "Topic set" to something like "Topic unset". Previously we'd see:

```
00:09 -!- Irssi: Join to #test was synced in 0 secs
00:09 -!- Topic unset by tester on #test
00:09 -!- Topic for #test:
00:09 -!- Topic set by tester [~tester@localhost] [Fri Aug 10 00:09:44 2018]
```

Maybe something like this would be good:

```
00:09 -!- Irssi: Join to #test was synced in 0 secs
00:09 -!- Topic unset by tester on #test
00:09 -!- No topic set for #test
00:09 -!- Topic unset by tester [~tester@localhost] [Fri Aug 10 00:09:44 2018]
```

But perhaps changing that text is overcomplicating things. I mean it is fairly uncommon for topics to be unset after all.

**Comment by horgh** [comment]
That PR has a fix for this. It doesn't do the revert & show when topic was unset though. That'd be fine too, so we could close it and go that way if we prefer.
